### PR TITLE
wallet-api: Base16 for hashes

### PR DESCRIPTION
**Pull request by j-mueller**

### CODE REVIEW

**Comment by j-mueller** [comment]
Can't seem to do the darwin builds :(

**Comment by michaelpj** [comment]
The darwin builders are in trouble. I'm happy that this is safe - if you mark it not as a draft I'll force-merge it.

File: wallet-api/src/Ledger/Types.hs [hunk 1]
```diff
@@ -197,16 +197,18 @@ instance Serialise (Digest SHA256) where
...
 
 instance FromJSON (Digest SHA256) where
   parseJSON = withText "SHA256" $ \s -> do
     let ev = do
-          eun64 <- Base64.decode . TE.encodeUtf8 $ s
-          first show $ deserialiseOrFail $ BSL.fromStrict eun64
+          let (eun16, rest) = Base16.decode . TE.encodeUtf8 $ s
+          if BSS.null rest
+          then first show $ deserialiseOrFail $ BSL.fromStrict eun16
+          else fail "failed to decode base16"
```
Comment (michaelpj) [inline wallet-api/src/Ledger/Types.hs]: The duplication of this code is annoying.

Comment (shmish111) [reply wallet-api/src/Ledger/Types.hs]: I agree, there must be a place we can put this

Comment (j-mueller) [reply wallet-api/src/Ledger/Types.hs]: OK

**Comment by j-mueller** [comment]
I think this is ready now.

**Comment by michaelpj** [comment]
Stylish haskell failure.

**Comment by j-mueller** [comment]
Just fixed that (https://hydra.iohk.io/eval/545993#tabs-now-succeed) - although I should really start running the script myself instead of getting the diffs from hydra :|

**Comment by michaelpj** [comment]
It should be better as of just recently - see my message in Slack. Won't try and rebuild rust now :D

**Pull request merged**